nautobot/core/nautobot_app/__init__.py

```diff
@@ -0,0 +1,169 @@
+from django.apps import AppConfig
+from django.urls import reverse
+from operator import getitem
+from collections import OrderedDict
+
+from nautobot.extras.choices import CustomLinkButtonClassChoices
+from nautobot.extras.plugins.utils import import_object
+from nautobot.extras.registry import registry
+
+
+class NautobotConfig(AppConfig):
+    menu_tabs = "navigation.menu_tabs"
+
+    def ready(self):
+        if self.__module__.endswith(".apps"):
+            self.__module__ = self.__module__[:-5]
+        menu_items = import_object(f"{self.__module__}.{self.menu_tabs}")
+        if menu_items is not None:
+            register_menu_items(menu_items)
+
+
+def register_menu_items(class_list):
+    for nav_tab in class_list:
+        registry["nav_menu"]["tabs"][nav_tab.name] = {
+            "weight": nav_tab.weight,
+            "groups": registry["nav_menu"]["tabs"].get(nav_tab.name, {}).get("groups", {}),
+            "permissions": [],
+        }
```

Comment:
This still reads very unintuitively to me, and it doesn't make sense to me for a nav tab's weight to be changeable after it gets created. Perhaps instead?
```suggestion
        if nav_tab.name not in registry["nav_menu"]["tabs"]:
            registry["nav_menu"]["tabs"][nav_tab.name] = {
                "weight": nav_tab.weight,
                "groups": {},
                "permissions": [],
            }
```

Reply:
I thought as one of the next tasks is to allow plugins to hide top level objects the tabs and the groups would need to be editable